networkit/cpp/distance/AffectedNodes.cpp

```diff
@@ -147,17 +147,19 @@ void AffectedNodes::addedEdge() {
     std::vector<edgeweight> levelImprovementU(numLevelsU + 1, 0);
     std::vector<edgeweight> levelImprovementV(numLevelsV + 1, 0);
 
-    for (edgeweight i = 1.0; i <= numLevelsV; i++) {
-      for (edgeweight j = 1.0; j <= numLevelsU; j++) {
-        levelImprovementV[i] += (numNodesOnLevelFromU[j] * 1.0 / (i + j)) -
-                                (prevNumNodesOnLevelFromU[j] * 1.0 / (i + j));
+    for (index i = 1.0; i <= numLevelsV; i++) {
+      for (index j = 1.0; j <= numLevelsU; j++) {
+          levelImprovementV[i] +=
+              (numNodesOnLevelFromU[j] * 1.0 / static_cast<double>(i + j))
+              - (prevNumNodesOnLevelFromU[j] * 1.0 / static_cast<double>(i + j));
```

Comment:
What was changed here?

Reply:
`edgeweight` -> `index`

Reply:
The warning is `clang-analyzer-security.FloatLoopCounter` [1], basically we should not use floats as loop counters.

[1]https://clang.llvm.org/docs/analyzer/checkers.html#security-floatloopcounter